src/lucky_migrator/create_table_statement.cr

```diff
@@ -113,6 +114,28 @@ class LuckyMigrator::CreateTableStatement
     raise "index on #{@table_name}.#{column} already exists"
   end
 
+  # Adds a references column and index given a model class and references option.
+  macro belongs_to(model_class, references = nil)
+    {% optional = model_class.is_a?(Generic) %}
```

Comment:
I *think* you can do this: `{% optional = model_class.stringify.ends_with?("?") %}` Note the call to `stringify` to get the string version of the class name

Reply:
I tried this originally but the problem I ran into is that User? is considered a Union between User and Nil, so it stringifies to "::Union(User, ::Nil)".

In a macro it's considered as a [Crystal::Macros::Generic](https://crystal-lang.org/api/0.23.1/Crystal/Macros/Generic.html) and you can get an array of the AST nodes using #type_vars, then stringify it from there without the "?".

https://play.crystal-lang.org/#/r/33qn

Reply:
Oh I see! I think it's more clear to use `model_class.is_a?(Union)`. That signifies that it is a union type and is what is used elsewhere in Lucky.